### PR TITLE
build: ssh connection middleware and authenticate middleware

### PR DESCRIPTION
**Pull request by Zaher1307**

### CODE REVIEW

File: src/middlewares/vm-connection.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const { NodeSSH } = require('node-ssh');
```
Comment (IslamWalid) [inline src/middlewares/vm-connection.js:1]: this file is better named with the name of the function it exposes, `vm-connect` is a better name.

File: src/middlewares/vm-connection.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  try {
+    await ssh.connect({
+      host,
+      username,
+      password
+    });
+    req.ssh = ssh;
+    next();
+  } catch (err) {
+    next(err);
```
Comment (IslamWalid) [inline src/middlewares/vm-connection.js:27]: you need - if applicable - to create an error handler module for any expected errors from the ssh connection.

File: src/utils/validate-credentials.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const { ResponseMessages } = require('./enums');
```
Comment (IslamWalid) [inline src/utils/validate-credentials.js:1]: I think it's better to use general input validator

**Review by meldisoukyy — APPROVED** [review]

File: src/middlewares/vm-connection.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+const { NodeSSH } = require('node-ssh');
+
+const validateCredentials = require('../utils/validate-credentials');
+const sendResErr = require('../utils/send-res-err');
+const { HttpStatus, ResponseMessages } = require('../utils/enums');
+
+const vmConnect = async (req, res, next) => {
+  const message = validateCredentials(req.body);
+  if (message) {
+    sendResErr(res, {
```
Comment (Eslam-Nawara) [inline src/middlewares/vm-connection.js:10]: You should return just after sending the response and not carry on executing the function.

Comment (Eslam-Nawara) [inline src/middlewares/vm-connection.js:10]: You should return just after sending the response and not carry on executing the function.

File: src/middlewares/vm-connection.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      status: HttpStatus.BAD_REQUEST,
+      message: ResponseMessages.MISSING_FIELDS
```
Comment (Eslam-Nawara) [inline src/middlewares/vm-connection.js:12]: It would be better to use the returned ` message ` constant instead of duplicating ` ResponseMessage...`

**Review by Eslam-Nawara — CHANGES_REQUESTED** [review]

**Review by Eslam-Nawara — APPROVED** [review]

**Pull request merged**